imports/plugins/core/ui-tagnav/client/containers/tagNavContainer.js

```diff
@@ -0,0 +1,392 @@
+    this.setState({
+      [NavbarStates.Visible]: isVisible,
+      tagIds,
+      tagsByKey });
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener("resize", this.onWindowResize);
+  }
+
+  onWindowResize = () => {
+    const matchQuery = window.matchMedia("(max-width: 991px)");
+    if (matchQuery.matches) {
+      this.setState({
+        [NavbarStates.Orientation]: NavbarOrientation.Vertical,
+        [NavbarStates.Position]: NavbarPosition.Fixed,
+        [NavbarStates.Anchor]: NavbarAnchor.Left
+      });
+    } else {
+      this.setState({
+        [NavbarStates.Orientation]: NavbarOrientation.Horizontal,
+        [NavbarStates.Position]: NavbarPosition.Static,
+        [NavbarStates.Anchor]: NavbarAnchor.None,
+        [NavbarStates.Visible]: false
+      });
+    }
+  }
+
+  handleNewTagSave = (tag, parentTag) => {
+    TagNavHelpers.onTagCreate(tag.name, parentTag);
+    this.setState({ newTag: { name: "" } });
+  }
+
+  handleNewTagUpdate = (tag) => { // updates the current tag state being edited
+    this.setState({
+      newTag: tag
+    });
+  }
+
+  handleTagRemove = (tag, parentTag) => {
+    TagNavHelpers.onTagRemove(tag, parentTag);
+  }
+
+  handleTagUpdate = (tag) => {
+    const newState = update(this.state, {
+      tagsByKey: {
+        [tag._id]: {
+          $set: tag
+        }
+      }
+    });
+
+    this.setState(newState);
+  }
+
+  handleTagSave = (tag) => {
+    TagNavHelpers.onUpdateTag(tag._id, tag.name);
+  }
+
+  handleMoveTag = (dragIndex, hoverIndex) => {
+    const tag = this.state.tagIds[dragIndex];
+
+    // Apply new sort order to variant list
+    const newState = update(this.state, {
+      tagIds: {
+        $splice: [
+          [dragIndex, 1],
+          [hoverIndex, 0, tag]
+        ]
+      }
+    });
+
+    // Set local state so the component does't have to wait for a round-trip
+    // to the server to get the updated list of variants
+    this.setState(newState, () => {
+      debounce(() => TagNavHelpers.onTagSort(this.state.tagIds), 500)(); // Save the updated positions
+    });
+  }
+
+  handleGetSuggestions = (suggestionUpdateRequest) => {
+    const suggestions = TagNavHelpers.updateSuggestions(
+      suggestionUpdateRequest.value,
+      { excludeTags: this.state.tagIds }
+    );
+
+    this.setState({ suggestions });
+  }
+
+  handleClearSuggestions = () => {
+    this.setState({ suggestions: [] });
+  }
+
+  get canEdit() {
+    return this.props.hasEditRights && Reaction.isPreview() === false;
+  }
+
+  attachBodyListener = () => {
+    document.body.addEventListener("mouseover", this.closeDropdown);
+    this.setState({ attachedBodyListener: true });
+  }
+
+  detachhBodyListener = () => {
+    document.body.removeEventListener("mouseover", this.closeDropdown);
+    this.setState({ attachedBodyListener: false });
+  }
+
+  closeDropdown = (event) => {
+    if ($(event.target).closest(".navbar-item").length === 0) {
```

Comment:
You can try using ref's `https://facebook.github.io/react/docs/refs-and-the-dom.html`. But yea, this there needs to be no jQuery otherwise we won't be able to use the navbar with SSR.